### PR TITLE
obs-browser: Fix key mapping on linux

### LINKED ISSUE #200: Interact window provides incorrect codes on keypress and keydown events
Whenever a `keydown` or `keypress` event gets triggered, `event.code` gets set to the wrong value.

## Expected Behavior

Add the following scripts and then hold down the A key when having any window selected. It should log `KeyA` for each of the events.

chrome_Fmef3ORunR

chrome_QmuZEl5Ii5

## Current Behavior

Add the following scripts and then hold down the A key when having the Browser Source Interact window selected. It logs `F7` and an empty string for each of the events.

chrome_kri1BEb3fG

chrome_exgsFEmOY1

## Steps to Reproduce

1. Start OBS with `--remote-debugging-port=8081` so you have access to the debug tools.
2. Create a new scene in OBS Studio, add the Browser source, and point it at the default web page.
3. Go to `http://localhost:8081`. Run the following script in the console:
```
document.addEventListener("keypress", event => {
  console.log(event.code);
});
```
4. Right click on the Browser source and click Interact.
5. Hold down the A key after selecting the Interact window. This results in the wrong key code being logged in the debug console.

## Additional information

`event.keyCode` produces the correct values, but `event.code` is broken for both the `keypress` and `keydown` events.

### PR DESCRIPTION
**Pull request by Francexi**
### Description
The fix maps, only for Linux systems, XKB keycodes (that are provided to the SendKeyClick function) to the Javascript ones. The fix also add the native_scancode as value for the native_vkey, that seems to make the fix work also on Windows. (thanks to WizardCM for the collab)

### Motivation and Context
The keypress event was inusable on obs-browser. This seems to fully fix this

Fixes #200 

### How Has This Been Tested?
Tested on Slackware-current, using CEF build 3809. As long as I tested, all seems to work fine

### Types of changes

Bug fix (non-breaking change which fixes an issue) 

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [ ] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.

### CODE REVIEW

**Comment by kkartaltepe** [comment]
Please add the appropriate library search in cmake along with adding the appropriate includes (but without linking since we do not actually need the library).

**Comment by kkartaltepe** [comment]
You can see a good example of a simple find module at
https://github.com/obsproject/obs-studio/blob/master/cmake/Modules/FindX11_XCB.cmake

Adding a module searching pkg-config for `x11`(case matters) will get you the PKG_X11_LIBRARIES and PKG_X11_INCLUDE_DIR and follow the pattern in the above file for filling out the variables.

Like with findCef just drop it in the root of this package https://github.com/obsproject/obs-browser/blob/master/FindCEF.cmake

Then in the CMakeLists.txt you can simply `find_package(X11)` and use the X11_INCLUDE_DIR in an `include_directories($X11_INCLUDE_DIR)` in some Linux specific `if` block.

**Comment by Francexi** [comment]
I don't know if this is right, I think is how it should work

**Comment by kkartaltepe** [comment]
libX11-xcb is separate from libX11. The module for it is an example that is relatively simple. You should replace xcb references with plain x11 references. And change the header being searched for from the xcb header `X11/Xlib-xcb.h` to the X11 header you are using `X11/keysym.h`.

You can run `cmake` to test it. If running cmake with this commit didnt fail on your system you must already have an existing FindX11.cmake in your system module path that found it.

**Comment by Francexi** [comment]
this seems to go well. I also passed the files through clang-format

File: CMakeLists.txt [hunk 1]
```diff
@@ -116,7 +126,7 @@ set(obs-browser_HEADERS
 
 # only allow browser panels on win32 for now -- other operating systems
 # need more testing
-if(WIN32 AND BROWSER_PANEL_SUPPORT_ENABLED)
+if(BROWSER_PANEL_SUPPORT_ENABLED)
```
Comment (kkartaltepe) [inline CMakeLists.txt]: This seems unrelated to the PR. Did you mean to add this change?

Comment (Francexi) [reply CMakeLists.txt]: woops, that was unintended. I corrected it

File: obs-browser-source.cpp [hunk 2]
```diff
@@ -290,17 +298,425 @@ void BrowserSource::SendFocus(bool focus)
...
+    case XF86XK_MonBrightnessDown:
+      return VKEY_BRIGHTNESS_DOWN;
+    case XF86XK_MonBrightnessUp:
+      return VKEY_BRIGHTNESS_UP;
+    case XF86XK_KbdBrightnessDown:
+      return VKEY_KBD_BRIGHTNESS_DOWN;
+    case XF86XK_KbdBrightnessUp:
+      return VKEY_KBD_BRIGHTNESS_UP;
+
+    // TODO(sad): some keycodes are still missing.
```
Comment (kkartaltepe) [inline obs-browser-source.cpp]: If you are going to copy and paste code from chromium at least document it so we know where to go to find updated versions.

For example this entire function is out of date with chromium head and noted that it only sort of works correctly on US layouts.
https://chromium.googlesource.com/chromium/src/+/lkgr/ui/events/keycodes/keyboard_code_conversion_x.cc

Comment (Francexi) [reply obs-browser-source.cpp]: well, that was the only mapping that I found that worked on both italian and US layouts (the two I had). If you want to suggest a modification in particular, can you do a pull request to my repo and I'll merge it
